library/acl.inc

```diff
@@ -828,7 +828,9 @@
             return false;
         elseif($count_user_allowed > 0)
             return true;
-        elseif($count_group_denied > 0)
+        // We do not exclude user that belongs to group A and B
+        // In case of A group have permissions and B group does not have permissions
+        elseif($count_group_denied > 0 && $count_group_allowed < 1)
```

Comment:
It is also possible that this zh function is not doing the correct query. I wonder if they are querying for a DENY or are just querying id doesn't have permission. If so, then need to fix the query, or else users in multiple groups would never pass.